cmake/ThirdParties.cmake

```diff
@@ -20,63 +22,57 @@ if(CONTOUR_TESTING OR CRISPY_TESTING OR LIBTERMINAL_TESTING)
 
   CPMAddPackage(
     NAME catch2
-    VERSION ${3rdparty_catch2_version}
-    URL https://github.com/catchorg/Catch2/archive/${3rdparty_catch2_version}.zip
-    URL_HASH SHA256=ab21f8c8804f3c9987d21a731440611d8dfecf8a80efcd2e46bf67a3d9bdf967
+    GITHUB_REPOSITORY "catchorg/Catch2"
+    GIT_TAG ${3rdparty_catch2_version}
     EXCLUDE_FROM_ALL YES
   )
 endif()
 
 CPMAddPackage(
   NAME fmt
-  VERSION ${3rdparty_fmt_version}
-  URL https://github.com/fmtlib/fmt/archive/${3rdparty_fmt_version}.zip
-  URL_HASH SHA256=d5a6f6e3431fec6332021dc69a14dd11b85b0103b6317d33b2e5e5a91fd08add
+  GITHUB_REPOSITORY "fmtlib/fmt"
+  GIT_TAG ${3rdparty_fmt_version}
   EXCLUDE_FROM_ALL YES
 )
 
 CPMAddPackage(
   NAME range_v3
-  VERSION ${3rdparty_range_v3_version}
-  URL https://github.com/ericniebler/range-v3/archive/${3rdparty_range_v3_version}.zip
-  URL_HASH SHA256=d866cb8f56aea491928af5cb1e7d2dbc4d46bd8a1a2a7b88194fb42b2c126e4b
+  GITHUB_REPOSITORY "ericniebler/range-v3"
+  GIT_TAG ${3rdparty_range_v3_version}
   EXCLUDE_FROM_ALL YES
 )
 
 set(YAML_CPP_BUILD_CONTRIB OFF CACHE INTERNAL "")
 set(YAML_CPP_BUILD_TESTS OFF CACHE INTERNAL "")
 set(YAML_CPP_BUILD_TOOLS OFF CACHE INTERNAL "")
 set(YAML_CPP_INSTALL OFF CACHE INTERNAL "")
+set(YAML_BUILD_SHARED_LIBS OFF CACHE INTERNAL "")
 
 CPMAddPackage(
   NAME yaml_cpp
-  VERSION ${3rdparty_yaml_cpp_version}
-  URL https://github.com/jbeder/yaml-cpp/archive/${3rdparty_yaml_cpp_version}.zip
-  URL_HASH SHA256=086e25ffc95eb54deb949f3e3343daff1722007375947da3d36b7ebd89e9ea97
+  GITHUB_REPOSITORY "jbeder/yaml-cpp"
+  GIT_TAG ${3rdparty_yaml_cpp_version}
   EXCLUDE_FROM_ALL YES
 )
 
 CPMAddPackage(
   NAME GSL
-  VERSION "3.1.0"
-  URL https://github.com/microsoft/GSL/archive/refs/tags/v3.1.0.zip
-  URL_HASH SHA256=a1041e41e60f9cb3789036f1c84ea9b4298823cbe94d16b096971fdc3de485b7
+  GITHUB_REPOSITORY "microsoft/GSL"
+  GIT_TAG ${3rdparty_gsl_version}
   EXCLUDE_FROM_ALL YES
 )
 
 CPMAddPackage(
   NAME termbenchpro
-  VERSION ${3rdparty_termbenchpro_version}
-  URL https://github.com/contour-terminal/termbench-pro/archive/${3rdparty_termbenchpro_version}.zip
-  URL_HASH SHA256=555d6dd4e3017e93cc3d6a0b0507e0e1a8eb4534d636ce6bf30ead2d8c0a9e0c
+  GITHUB_REPOSITORY "contour-terminal/termbench-pro"
+  GIT_TAG ${3rdparty_termbenchpro_version}
   EXCLUDE_FROM_ALL YES
 )
 
 CPMAddPackage(
   NAME libunicode
-  VERSION ${3rdparty_libunicode_version}
-  URL https://github.com/contour-terminal/libunicode/archive/${3rdparty_libunicode_version}.zip
-  URL_HASH SHA256=1884993f33147121660cdc0df50d2d7b5fc8c699a01ea83a3ff7b16409893989
+  GITHUB_REPOSITORY "contour-terminal/libunicode"
+  GIT_TAG ${3rdparty_libunicode_version}
```

Comment:
The use of `GITHUB_REPOSITORY` and `GIT_TAG` is something I actually quite like. We should have done this earlier. :+1: